scene/resources/animation.cpp

```diff
@@ -4233,14 +4246,19 @@ void Animation::_position_track_optimize(int p_idx, real_t p_allowed_velocity_er
 	ERR_FAIL_INDEX(p_idx, tracks.size());
 	ERR_FAIL_COND(tracks[p_idx]->type != TYPE_POSITION_3D);
 	PositionTrack *tt = static_cast<PositionTrack *>(tracks[p_idx]);
-
+	bool is_nearest = false;
+	if (tracks[p_idx]->interpolation == INTERPOLATION_NEAREST) {
+		is_nearest = true;
+	}
+	else if (tracks[p_idx]->interpolation != INTERPOLATION_LINEAR) {
+		return;
+	}
```

Comment:
```suggestion
	bool is_nearest = false;
	if (tracks[p_idx]->interpolation == INTERPOLATION_NEAREST) {
		is_nearest = true;
	}
	else if (tracks[p_idx]->interpolation != INTERPOLATION_LINEAR) {
		return;
	}
	PositionTrack *tt = static_cast<PositionTrack *>(tracks[p_idx]);
```
It is better to place the checking just below the error check because static_cast may be ignored.